src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -207,6 +207,23 @@ public function findOneByIdAndChannel($id, ChannelInterface $channel = null)
     /**
      * {@inheritdoc}
      */
+    public function findByTaxonCodeAndChannel($code, ChannelInterface $channel)
```

Comment:
Yes, it is disabled permanently. ``enabled`` flag indicates that. Being out-of-stock, current unavailable, discontinued are separate features and will be handled differently.